prov/rxm/src/rxm_ep.c

```diff
@@ -1250,27 +1290,31 @@ rxm_ep_send_common(struct rxm_ep *rxm_ep, const struct iovec *iov, void **desc,
 	if (data_len <= rxm_ep->rxm_info->tx_attr->inject_size) {
 		size_t total_len = sizeof(struct rxm_pkt) + data_len;
 
-		ret = rxm_ep_format_tx_res_lightweight(
+		if ((flags & FI_INJECT) && !(flags & FI_COMPLETION) &&
+		    (total_len <= rxm_ep->msg_info->tx_attr->inject_size)) {
+			ret = rxm_ep_format_tx_res_lightweight(
 					rxm_ep, rxm_conn, data_len, data,
 					flags, tag, &tx_buf,
-					&rxm_ep->buf_pools[RXM_BUF_POOL_TX]);
+					&rxm_ep->buf_pools[RXM_BUF_POOL_TX_INJECT]);
```

Comment:
Just curious. Did you by any chance check if using alloca or having a local stack variable for tx buffer gave better performance than getting the buffer from pool for this case?

Reply:
Yes, I tried this approach, but this increase latency on ~0.01. Instructions count is reduced on ~20 in the fi_tinject() by this approach.